src/load_balancer/load_balancer.rs

```diff
@@ -9,23 +10,31 @@ use hyper::{
 };
 use hyper_util::rt::TokioIo;
 use tokio::net::TcpStream;
+use tokio::sync::RwLock;
 
 use crate::data_store::Worker;
-use crate::utils::{validate_address, DF_TABLE_NAME};
+use crate::utils::{df_to_table, validate_address, DF_TABLE_NAME};
 use crate::BoxBody;
 use crate::error::LoadBalancerError;
 
 // load balancer now holds ctx with registered table with worker hosts information
 pub struct LoadBalancer {
     pub ctx: SessionContext,
     pub current_worker: usize,
+    pub algorithm: Algorithm,
+}
+
+pub enum Algorithm {
+    RoundRobin, // Distribute requests evenly across all worker servers
+    LeastConnections, // Route requests to the worker server with the least active connections.
 }
 
 impl LoadBalancer {
-    pub fn new(ctx: SessionContext, current_worker: Option<usize>) -> Result<Self, LoadBalancerError> {
+    pub fn new(ctx: SessionContext, current_worker: Option<usize>, algorithm: Option<Algorithm>) -> Result<Self, LoadBalancerError> {
```

Comment:
Can we expect to get a value for current_worker? I would like to know what scenario that `None` could be passed into this function?

Reply:
Maybe you're write and it's better to make it more verbose and clear for user.